src/main/java/task/Task.java

```diff
@@ -0,0 +1,43 @@
+package task;
+
+import java.time.LocalDate;
+
+public class Task {
+    protected String task;
+    protected boolean done;
+
+    public Task(String task) {
+        this.task = task;
+        this.done = false;
+    }
+
+    public Task(String task, boolean done) {
+        this.task = task;
+        this.done = done;
+    }
+
+    public Task mark() {
+        return new Task(task, true);
+    }
+
+    public Task unmark() {
+        return new Task(task, false);
+    }
```

Comment:
Not too sure about the rationale behind the need to `new()` an object, (same for `Deadline` `Event` and `Todo`), why not just simply set the `done` field as true or false, all the children of `Task` share the same behaviour and they can just access `mark()` and `unmark()` straight away.